cmd/earthly/main.go

```diff
@@ -1278,6 +1280,44 @@ func (app *earthlyApp) processDeprecatedCommandOptions(context *cli.Context, cfg
 	return nil
 }
 
+func (app *earthlyApp) unhideFlags(ctx context.Context) error {
+	var err error
+	showHidden := strings.HasPrefix(Version, "dev-")
```

Comment:
Part of me thinks we should keep these flags hidden so we don't get confused with what is hidden or not in the actual released versions. The `EARTHLY_SHOW_HIDDEN` would be there to always show it otherwise

Reply:
If we run `EARTHLY_SHOW_HIDDEN=0 earthly` it'll force them to be hidden even when running a dev version. When `EARTHLY_SHOW_HIDDEN` isn't set, we default to this auto behaviour, and `EARTHLY_SHOW_HIDDEN=1` we will always show them.

I have `EARTHLY_SHOW_HIDDEN=1` set in my `.bashrc`, so I don't have a strong opinion on removing the auto heuristic.

Reply:
@andrewsykim it's been removed. Can you take another look?